PIC32MX4 Modifications & MPLAB Project

n9wxu opened the pull request:
Improved performance and ported to the MX440.
Temporarily removed the PWM feature.
Added additional inputs to support a derivative HW board.
Increased the step rate to 138khz.  80Mhz is starting to run out of steam at this point.
Getting ready to use the PIC32MX470 to increase the clock speed to 120MHz.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kinsamanka/picnc-v2/4)
<!-- Reviewable:end -->

n9wxu committed:
PIC32MX4 Modifications & MPLAB Project

Improved performance and ported to the MX4.

n9wxu committed:
Actual changes to main and stepgen

Previous update did not include the important changes.

n9wxu commented:
Forgot to include the important changes to main & stepgen.

n9wxu committed:
Added the V3 HW files

These files are a modification from the OSHPARK build.
OSHPARK revision needs MCLR & I2C pullup resistors.  I2C is optional for
now.

n9wxu commented:
Hardware files for the V3 PI Hat.
V3 hardware is here https://oshpark.com/shared_projects/xTIz9hC0
The oshpark build is missing a MCLR pullup resistor.

n9wxu committed:
X1_ENABLE  is not connected on the Hardware

PWM's now mapped to the step channels.
Enables are now mapped as the step channels
E1_Direction is now the master stepper driver enable.
X_STEP is still the step because X_ENABLE is defective.
OC2-4 are used for PWM OC1 is not used.

n9wxu commented:
Some additional hardware issues on the V3 board at OSHPARK.
X1_ENABLE is not connected to the PIC32.  X1_Step will remain unchanged.  Other Enables will become step generators and the step generators will become PWM.

adbensi commented:
very interesting, I'm changing the board for debugging and testing .. congradulations n9wxu!
![top](https://cloud.githubusercontent.com/assets/8729496/11915373/4233d758-a68b-11e5-9c91-2912e07a13ef.png)
![bot](https://cloud.githubusercontent.com/assets/8729496/11915374/423513ac-a68b-11e5-94fd-a51e08c7ea82.png)

n9wxu commented:
Good looking board.  My only comment is it is generally a bad idea to put via's under pads.  The reason is due to solder stencils.  The via will absorb more solder than the pad will so you need a larger opening in the stencil to put more paste on those pads.  It is better to move the via away from the pad so all stencil openings are a consistent size.  Obviously if you do not plan on using stencils, this is not an issue.

n9wxu commented:
![img_1665](https://cloud.githubusercontent.com/assets/750641/11915618/d7febb2e-a66b-11e5-83ff-1682304f46f1.png)
![img_1666](https://cloud.githubusercontent.com/assets/750641/11915619/d81ddd42-a66b-11e5-83f7-4bc1fdad2200.png)

Board on right has engineering sample MX470. That will run at 120Mhz.

n9wxu committed:
200khz update

refactored to go back to a single call to stepgen.
added a macro to make the step generation DRY.
reduced the step accumulator to 65536 (not really sure if this is a good
idea)
increased the core timer rate to 200khz (52% CPU for step generation on
4 Axis)
Made the appropriate changes to the HAL.
programmable stepwidth is still not supported.

n9wxu commented:
Reached 200KHz.  The stepper motor really moves!

n9wxu committed:
restored original step algorithm.

I restored the original step algorithm sans stepwidth adjustment.  It is
still operating at 200khz.

adbensi commented:
n9wxu, You are using SYS_FREQ = 80MHz and BASEFREQ = 150KHz with MX470 or MX440?
great job! I look forward to see these speeds :+1:

kinsamanka commented on HAL/picnc.h:
```diff
@@ -50,11 +50,11 @@ typedef enum {
 #define BUFSIZE			(SPIBUFSIZE/4)
 
-#define STEPBIT			22		/* bit location in DDS accum */
+#define STEPBIT			16		/* bit location in DDS accum */
```
You are sacrificing step resolution over speed

n9wxu replied:
The max speed step frequency is not limited by those settings. In any case I put your algorithm back. Stand by for another pull request. 

Sent from my iPhone

> On Dec 21, 2015, at 3:51 PM, GP Orcullo notifications@github.com wrote:
> 
> In HAL/picnc.h:
> 
> > @@ -50,11 +50,11 @@ typedef enum {
> >  #define SPIBUFSIZE     20      /\* SPI buffer size */
> >  #define BUFSIZE            (SPIBUFSIZE/4)
> > 
> > -#define STEPBIT            22      /\* bit location in DDS accum _/
> > +#define STEPBIT            16      /_ bit location in DDS accum */
> > You are sacrificing step resolution over speed
> 
> —
> Reply to this email directly or view it on GitHub.

n9wxu commented:
I push more code up tonight because it is at 200khz now. 

Sent from my iPhone

> On Dec 21, 2015, at 3:43 PM, adbensi notifications@github.com wrote:
> 
> n9wxu, You are using SYS_FREQ = 80MHz and BASEFREQ = 150KHz with MX470 or MX440?
> great job! I look forward to see these speeds 
> 
> —
> Reply to this email directly or view it on GitHub.

n9wxu commented:
This restores the higher resolution steps and keeps the 200khz step rate.

n9wxu committed:
FIFO output attempt

attempted to speed up the ISR by generating the step pattern in a FIFO

n9wxu committed:
put it back to 200khz ISR generated steps

It can't produce more than 1 step ahead reliably so the ISR runs dry
when operating with a FIFO.

adbensi commented:
n9wxu, if You have odroid C1, there is last RT kernel
https://github.com/adbensi/kernel-odroidc-3.10.80-rt102
Best Regards

adbensi commented:
![20160402_140117](https://cloud.githubusercontent.com/assets/8729496/14235538/badadfb8-f9d7-11e5-9f3d-6e4644746a2a.jpg)
 Odroid C0 ready :D

kinsamanka commented:
The Orange Pi One is a cheaper alternative ;-)
On Apr 4, 2016 6:08 AM, "adbensi" notifications@github.com wrote:

> [image: 20160402_140117]
> https://cloud.githubusercontent.com/assets/8729496/14235538/badadfb8-f9d7-11e5-9f3d-6e4644746a2a.jpg
> Odroid C0 ready :D
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/kinsamanka/PICnc-V2/pull/4#issuecomment-205074219

creisey commented:
how is the state at that project? im looking forwards for this

n9wxu commented:
I have had time for this project in some time. 
If I was starting now, I would use a teensy 4.0  the higher clock speed would speed the step rate considerably.

Joseph Julicher

> On Mar 7, 2020, at 9:06 AM, iosias <notifications@github.com> wrote:
> 
> ﻿
> how is the state at that project? im looking forwards for this
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.

creisey commented:
> ![20160402_140117](https://cloud.githubusercontent.com/assets/8729496/14235538/badadfb8-f9d7-11e5-9f3d-6e4644746a2a.jpg)
> Odroid C0 ready :D

i have some trouble with getting the odroid c0 running with linuxcnc... how did you proceed? all the tutorials i found are more than 2 year old and seam not woring anymore
with the prebuilt machinkit image from https://0ptr.link i have trouble with SPI and network

cheers

adbensi commented:
Hi!

 You need my kernel changed to use ethernet on odroid.. please, try on my
Github

Best regards

Em ter, 10 de mar de 2020 04:36, iosias <notifications@github.com> escreveu:

> [image: 20160402_140117]
> <https://cloud.githubusercontent.com/assets/8729496/14235538/badadfb8-f9d7-11e5-9f3d-6e4644746a2a.jpg>
> Odroid C0 ready :D
>
> i have some trouble with getting the odroid c0 running with linuxcnc...
> how did you proceed? all the tutorials i found are more than 2 year old and
> seam not woring anymore
> with the prebuilt machinkit image from https://0ptr.link i have trouble
> with SPI and network
>
> cheers
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kinsamanka/PICnc-V2/pull/4?email_source=notifications&email_token=ACCTHGG3NAQMIDG6557BJCTRGXUXHA5CNFSM4BW7RGW2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEOKKR6Y#issuecomment-596945147>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCTHGFTKHAOWKUXMUXMPIDRGXUXHANCNFSM4BW7RGWQ>
> .
>

creisey commented:
i tried that, but i had problems during the make process.

![image](https://user-images.githubusercontent.com/19647655/76326873-8bc3fa00-62e9-11ea-8e63-43f8076c5fa1.png)
it seemed related to gcc version 7 

could you provide a compiled kernel to me?
and what bases did you use? the Ubuntu MATE 18.04.3 image form hardkernel?

adbensi commented:
Hi,

In January 2016, I compiled directly on the Odroid C0, in one images from
repo Odroid that year .. this is a good reference to follow the same.

But Guys, recently I saw messages that there is something more recent here
with Tennsy, it is working too?

Em ter., 10 de mar. de 2020 às 12:10, iosias <notifications@github.com>
escreveu:

> i tried that, but i had problems during the make process.
>
> [image: image]
> <https://user-images.githubusercontent.com/19647655/76326873-8bc3fa00-62e9-11ea-8e63-43f8076c5fa1.png>
> it seemed related to gcc version 7
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kinsamanka/PICnc-V2/pull/4?email_source=notifications&email_token=ACCTHGEIDZT7H3AS67KV37LRGZJ6HA5CNFSM4BW7RGW2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEOLZ6XI#issuecomment-597139293>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCTHGBZT2ODMSYITL5YDYDRGZJ6HANCNFSM4BW7RGWQ>
> .
>

creisey commented:
i tried that, but i still cant compile, there are issues with gcc7, do you have the Odroid c0 running with Picnic?

is there a chance to buy a picnic hat? tindie option? ;)
